File: 프로그래머스/2/176962. 과제 진행하기/과제 진행하기.js
function changeTime(time) {
    time = time.split(":")
    return +time[0] * 60 + +time[1]
}

function solution(plans) {
  const answer = [];

  plans = plans
    .map(([name, start, playtime]) => [name, changeTime(start), +playtime])
    .sort((a, b) => b[1] - a[1]);

  while (plans.length) {
    const [name, start, playtime] = plans.pop();

    answer.forEach((v, i) => {
      if (start < v[1]) answer[i][1] += playtime;
    });
      
    answer.push([name, start + playtime]);
  }
    
  return answer.sort((a, b) => a[1] - b[1]).map(v => v[0]);
}

File: 프로그래머스/2/176962. 과제 진행하기/과제 진행하기.py
def change_time(time):
    time = time.split(":")
    return int(time[0]) * 60 + int(time[1])

def solution(plans):
    answer = []

    plans = [[name, change_time(start), int(playtime)] for name, start, playtime in plans]
    plans.sort(key=lambda x: x[1], reverse=True)

    while plans:
        name, start, playtime = plans.pop()

        for i in range(len(answer)):
            if start < answer[i][1]: answer[i][1] += playtime

        answer.append([name, start + playtime])

    answer.sort(key=lambda x: x[1])
    
    return [v[0] for v in answer]
